Sources/SourceKitLSP/DocumentManager.swift

```diff
@@ -150,13 +150,13 @@ public final class DocumentManager {
           
           // Remove all tokens in the updated range and shift later ones.
 
-          let previousLineCount = 1 + range.upperBound.line - range.lowerBound.line
+          let replacedLineCount = 1 + range.upperBound.line - range.lowerBound.line
           let newLines = edit.text.split(separator: "\n", omittingEmptySubsequences: false)
-          let lastLineReplaceLength = (
-            range.lowerBound.line == range.upperBound.line ? range.upperBound.utf16index : 0
-          ) - range.lowerBound.utf16index
-          let lastLineLengthDelta = newLines.last!.count - lastLineReplaceLength
-          let lineDelta = newLines.count - previousLineCount
+          let upperUtf16IndexAfterEdit = (
+            newLines.count == 1 ? range.lowerBound.utf16index : 0
+          ) + newLines.last!.utf16.count
+          let charDelta = upperUtf16IndexAfterEdit - range.upperBound.utf16index
```

Comment:
Nitpick: Isn’t this more of a `lastLineCharDelta`? `charDelta` implies to me that it should be applied in the same locations that `lineDelta` is applied.